MaxText/layers.py

```diff
@@ -609,11 +609,22 @@ def __call__(self, inputs, decode: bool = False, deterministic: bool = False):
 
     # Take elementwise product of above intermediate activations.
     x = functools.reduce(operator.mul, activations)
-    # Apply dropout and final dense output projection.
-    x = nn.Dropout(
-        rate=self.intermediate_dropout_rate, broadcast_dims=(-2,))(
-            x, deterministic=deterministic)  # Broadcast along length.
-    x = nn.with_logical_constraint(x, ('activation_batch', 'activation_length', 'activation_mlp'))
+
+    up_proj_x = DenseGeneral(
+          self.intermediate_dim,
+          dtype=self.dtype,
+          kernel_init=self.kernel_init,
+          kernel_axes=('embed', 'mlp'),
+          name="ffn_layer1",
+          config=cfg)(
+              inputs)
+    x = jnp.multiply(x, up_proj_x)
+
+    # No Dropout in Llama
+    # x = nn.Dropout(
+    #     rate=self.intermediate_dropout_rate, broadcast_dims=(-2,))(
+    #         x, deterministic=deterministic)  # Broadcast along length.
+    # x = nn.with_logical_constraint(x, ('activation_batch', 'activation_length', 'activation_mlp'))
```

Comment:
I think we still need this sharding constraint.

Reply:
Brought it back - thanks!